src/pint/gridutils.py

```diff
@@ -900,264 +900,3 @@ def tuple_chisq_derived(
-        # ii indexes rows, now for have each column done by a different process
-        proc = Process(
-            target=_grid_docol, args=(ftr, par1_name, par1, par2_name, par2_grid, ii, q)
-        )
-        processes.append(proc)
-
-    # Now consume the list of processes by starting up to ncpu processes at a time
-    while len(processes):
-        # Start up to ncpu processes
-        started = []
-        for cpunum in range(ncpu):
-            if len(processes):
-                proc = processes.pop()
-                proc.start()
-                started.append(proc)
-        # Collect all the results from the started processes
-        for proc in started * len(par2_grid):
-            ii, jj, ch = q.get()
-            # Array index here is rownum, colnum so translates to y, x
-            chi2[jj, ii] = ch
-
-        # Now join each of those that are done to close them out
-        # This will be inefficient if the processes have large differences in runtime, since there
-        # is a synchronization point. In this case it should not be a problem.
-        for proc in started:
-            proc.join()
-            # print("|", end="")
-        # print("")
-
-    return chi2
-
-
-def old_grid_chisq(ftr, par1_name, par1_grid, par2_name, par2_grid):
-    """Compute chisq over a grid of two parameters, serial version
-    Single-threaded computation of chisq over 2-D grid of parameters.
-    Parameters
-    ----------
-    ftr
-        The base fitter to use.
-    par1_name : str
-        Name of the first parameter to grid over
-    par1_grid : array, Quantity
-        Array of par1 values for column of the output matrix
-    par2_name : str
-        Name of the second parameter to grid over
-    par2_grid : array, Quantity
-        Array of par2 values for column of the output matrix
-    Returns
-    -------
-    array : 2-D array of chisq values with par1 varying in columns and par2 varying in rows
-    """
-
-    # Save the current model so we can tweak it for gridding, then restore it at the end
-    savemod = ftr.model
-    gridmod = copy.deepcopy(ftr.model)
-    ftr.model = gridmod
-
-    # Freeze the two params we are going to grid over
-    getattr(ftr.model, par1_name).frozen = True
-    getattr(ftr.model, par2_name).frozen = True
-
-    # All other unfrozen parameters will be fitted for at each grid point
-
-    chi2 = np.zeros((len(par1_grid), len(par2_grid)))
-    # Want par1 on X-axis and par2 on y-axis
-    for ii, par1 in enumerate(par1_grid):
-        getattr(ftr.model, par1_name).quantity = par1
-        for jj, par2 in enumerate(par2_grid):
-            getattr(ftr.model, par2_name).quantity = par2
-            # Array index here is rownum, colnum so translates to y, x
-            chi2[jj, ii] = ftr.fit_toas()
-            # print(".", end="")
-        # print("")
-
-    # Restore saved model
-    ftr.model = savemod
-    return chi2
-
-
-##############################
-
-
-def plot_grid_chisq(
```

Comment:
This function is mentioned in `__all__` of this module. Shouldn't that be removed?

Reply:
Is there a different grid plotting function that replaces this one?

Reply:
Is https://nanograv-pint--1443.org.readthedocs.build/en/1443/_autosummary/pint.gridutils.grid_chisq.html#pint.gridutils.grid_chisq a useful example?

Reply:
I tried running the examples from the docs above. They work for me. The 2D example is missing a `plt.show()` at the end, although that's not a big deal.

I think the whole thing is slowed down significantly by logging. It would be nice to temporarily turn off logging for applications like this where some process is run in a loop.